tests/admin_filters/tests.py

```diff
@@ -1099,3 +1104,43 @@ def test_list_filter_queryset_filtered_by_default(self):
         changelist = modeladmin.get_changelist_instance(request)
         changelist.get_results(request)
         self.assertEqual(changelist.full_result_count, 4)
+
+    def test_should_show_date_hierarchy_choices(self):
+        modeladmin = BookAdminWithDateHierarchy(Book, site)
+
+        Book.objects.all().delete()
+        for date_registered in (
+            datetime.date(2017, 10, 1),
+            datetime.date(2017, 10, 1),
+            datetime.date(2017, 12, 15),
+            datetime.date(2017, 12, 15),
+            datetime.date(2017, 12, 31),
+            datetime.date(2018, 2, 1),
+        ):
+            Book.objects.create(title='title', date_registered=date_registered)
+
+        def _test_date_hierarchy_choices(query, expected_choices):
```

Comment:
Instead of helper method like this, you can use a loop and `subTest()`. See f04e6732c33e0fd67e82ab2d59f230fa0045d15b for an example.